src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] tables;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        tables = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        updateThreshold();
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int bucketIndex = getBucketIndex(key);
+        if (tables[bucketIndex] == null) {
+            tables[bucketIndex] = newNode;
+            size++;
+        } else {
+            linkNode(tables[bucketIndex], newNode);
+        }
+        if (size >= threshold) {
+            resizeTables();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> current = tables[getBucketIndex(key)];
+        while (current != null) {
+            if (current.equalsByKey(key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return Math.abs(Objects.hashCode(key));
+    }
+
+    private int getBucketIndex(K key) {
+        return hash(key) % tables.length;
+    }
+
+    private void linkNode(Node<K, V> current, Node<K, V> newNode) {
+        while (true) {
+            if (current.equalsByKey(newNode.key)) {
+                current.value = newNode.value;
+                return;
+            }
+            if (!current.hasNext()) {
+                current.next = newNode;
+                size++;
+                return;
+            }
+            current = current.next;
+        }
```

Comment:
I think we can simplify this loop. First of all it's not a good practice to make infinite loop. Secondly, you can connect node not only to the end. For example, if we have index we can say:
 `table[inedex] = new Node<>(hash, key, value, table[index]);`, so you don't need do it in loop.
```suggestion
        while (current != null) {
            if (current.equalsByKey(newNode.key)) {
                current.value = newNode.value;
                return;
            }
            current = current.next;
        }
        
        // connect outside loop
```